### PR TITLE
feat(mapi-v2): add parentId query param

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-3043

## Description

Add query param to /pages endpoint for user to search by parentId to return a list of pages with the given parentId.
Add breadcrumb to response.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-vtksdfnpuy.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 1]
```diff
@@ -1629,16 +1629,22 @@ paths:
...
         get:
+            parameters:
+                - $ref: "#/components/parameters/parentId"
             tags:
                 - API Documentation
             summary: Get API documentation pages
             description: |-
-                Get API documentation pages. The result contains a list of pages and folders of the API.
+                Get API documentation pages.
+                When no parameters are specified, returns all the pages of the API.
```
Comment (loriepisicchio) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml]:
```suggestion
                When no parentId is specified, returns all the pages of the API.
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 2]
```diff
@@ -1629,16 +1629,22 @@ paths:
...
+                
+                When parentId is specified (either ID of a folder, or 'ROOT'), only the pages with the given parent are returned, 
+                and the breadcrumb of the specified folder is added to the response. 
                 
                 User must have the API_DOCUMENTATION[READ] permission.
             operationId: getApiPages
             responses:
-                "200":
+                "201":
```
Comment (loriepisicchio) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml]:
```suggestion
                "200":
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiGetDocumentationPagesUsecase.java [hunk 3]
```diff
@@ -15,23 +15,55 @@
...
-    public record Input(String apiId) {}
+    public record Input(String apiId, String parentId) {}
+
+    public record Output(List<Page> pages, List<Breadcrumb> breadcrumbs) {}
 
-    public record Output(List<Page> pages) {}
+    private LinkedList<Page> constructBreadcrumbs(String parentId, LinkedList<Page> breadcrumbs) {
+        if (!StringUtils.isEmpty(parentId)) {
+            try {
+                final Page optParent = pageCrudService.get(parentId);
```
Comment (loriepisicchio) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiGetDocumentationPagesUsecase.java]: Suggestion: add a method `find` in pageCrudService that doesn't throw exception but instead returns an Optional.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 4]
```diff
@@ -6100,6 +6142,11 @@ components:
                                 type: array
                                 items:
                                     $ref: "#/components/schemas/Page"
+                            breadcrumb:
```
Comment (loriepisicchio) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:6145]:
```suggestion
                            breadcrumbs:
```
WDYT ?

Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:6145]: The ensemble of elements create a breadcrumb, so I would keep it singular.

**Review by loriepisicchio — APPROVED** [review]

**Pull request merged**